lib/src/combined_wrappers/combined_iterable.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+
+/// Returns a new iterable that represents iterables accessed sequentially.
+///
+/// All methods and accessors treat the new iterable as-if it were a single
+/// sequence, but the underlying implementation is based on lazily accessing
+/// individual iterable instances.
+class CombinedIterableView<T> extends IterableBase<T> {
```

Comment:
It's probably worth special-casing the `length` getter here, since many iterables are able to compute it more efficiently than by doing a full iteration. Same for `isEmpty`.